scripts/lint.sh

```diff
@@ -18,6 +18,10 @@ let "exitcode |= $?"
 grep -rn 'DISABLED_' src/test | grep -v '#[0-9]\+' | sed 's/^\([a-zA-Z/._]*:[0-9]*\).*/\1  Disabled tests should be documented with their issue number (e.g. \/* #123 *\/)/'
 let "exitcode |= $?"
 
+# The singleton pattern should not be manually implemented
+grep -rn 'static[^:]*instance;' --exclude src/lib/utils/singleton.hpp src | grep -v '#[0-9]\+' | sed 's/^\([a-zA-Z/._]*:[0-9]*\).*/\1  Singletons should not be implmented manually. Have a look at src\/utils\/singleton.hpp/'
```

Comment:
Na, and if I call my singleton with `get()?` I'm all for linting and stuff, but this we should probably just look out for in reviews.

Reply:
If it makes it through the review, we have a big problem though because the effects of the singleton problem are hard to isolate.

Reply:
My point is just that this is very error prone to check specifically for Singletons returned by a function with `instance` in the name, when there are many other ways that such a singleton-returning function could be named. So, this step in the lint script mostly provides a wrong sense of security and adds additional bash-noise and bash-noise is never good. Just my 2ct, keep it if you like it.

Reply:
What do you mean with calling a singleton with `get()`? This is not about having `instance` in the function name, but checking for a static variable called instance (which is not a single bit better):

```
static MyType& get() {
    static MyType instance;
    return instance;
}
```

However, I think it is better than nothing, but I also don't mind to remove it. Feel free to write a more thorough check.